models/Asset/Image/Thumbnail/Processor.php

```diff
@@ -335,11 +335,20 @@ public static function process(
                 }
 
                 $tmpFsPath = File::getLocalTempFilePath($fileExtension);
-                $image->save($tmpFsPath, $format, $config->getQuality());
-                $stream = fopen($tmpFsPath, 'rb');
-                $storage->writeStream($storagePath, $stream);
-                if (is_resource($stream)) {
-                    fclose($stream);
+
+                $fileHandle = null;
+
+                if ($format === 'original') {
+                    $fileHandle = fopen($asset->getLocalFile(), 'rb');
```

Comment:
Mhhh, haven't reproduce it yet but just casually noticed by looking at the code that when using Imagick, it wouldn't get to this point and the GIF would be always animated (`ORIGINAL` (.gif) and `PreserveAnimation:false`)
https://github.com/pimcore/pimcore/blob/3ffaeac8d9f07f877ea4f469015be1774de447b9/lib/Image/Adapter/Imagick.php#L267-L268

Could you please have a second look if it's doing like that?